### PR TITLE
Returning callback results when calling pipelines' train method

### LINKED ISSUE #62: Loss value history
**Is your feature request related to a problem? Please describe.**
I can't seem to find documentation or an object that holds the training loss history.

**Describe the solution you'd like**
Training history is saved in a logger, but not in a structured python object from which one can make basic loss curve plots. I would like to have an output from the `train` method of a trainer, or from a pipeline that allows me to plot basic loss curves.

**Describe alternatives you've considered**
In the BaseTrainer one could have (I think, I'm not sure how to do it) a callback that saves this basic information in an object.
**Additional context**
My code is taken from the examples:
```python
# Set up the training configuration
my_training_config = BaseTrainerConfig(
    output_dir='my_model',
    num_epochs=100,
    learning_rate=1e-3,
    batch_size=256,
    steps_saving=None
)
# Set up the model configuration
my_vae_config = model_config = VAEConfig(
    input_dim=(data.shape[1],),
    latent_dim=10
)
# Build the model
my_vae_model = VAE(
    model_config=my_vae_config,
    decoder=DecoderVAE(args=my_vae_config))
# Build the Pipeline
pipeline = TrainingPipeline(
    training_config=my_training_config,
    model=my_vae_model
)
# Launch the Pipeline
pipeline(
    train_data=data, # must be torch.Tensor or np.array
    eval_data=data[:1024] # must be torch.Tensor or np.array
)
```
As far as I know, there's no way of passing a callback to the training pipeline.

### PR DESCRIPTION
**Pull request by VolodyaCO**
Closes #62.

This PR is a proof of concept for returning values from callbacks which might be useful for immediate manipulation after the pipeline has been run.

### CODE REVIEW

**Comment by clementchadebec** [comment]
Hi @VolodyaCO, sorry to come back to you so late. 
Thank you very much for this contribution. 

I was wondering though if we should not save the metrics in an easily readable file such as a `csv` instead of storing them in a `dict` and returning them at the end of the training as you suggest. The other `TrainingCallbacks` do not indeed return anything at the `on_train_end` stage and thus neither does `trainer.train()` in most cases. Therefore, I wonder if this will create confusion for users. 

Let me know what you think of this :)

**Comment by VolodyaCO** [comment]
No worries @clementchadebec . I prefer not having side-effects such as saving things to a file. However, I see that to keep the changes consistent, one would want other `TrainingCallbacks` to return something explicitly (and at the moment they return None).

It looks as though another simple solution is not to return anything and not to save anything to a file. When using such a callback, the user would access the callback history and that's it. What do you think?

**Comment by clementchadebec** [comment]
Hi @VolodyaCO, thanks for the suggestion.
I agree with you that this seems to be the best solution to keep things consistent since the metrics would be available in the `history` attribute anyway.

**Comment by VolodyaCO** [comment]
@clementchadebec sorry for the delay. I just removed the returns!

**Review by clementchadebec — APPROVED** [review]
Hi @VolodyaCO,
No worries!
Thank you so much for making these changes. Everything looks fine :)
